### PR TITLE
Add changelog for 2.14.0 release

### PR DESCRIPTION
**Pull request by jonathanmos**
### What does this PR do?
• Promote 2.14.0 to release
• Update changelog

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,32 @@
+# 2.14.0 / 2024-09-24
```
Comment (0xnm) [inline CHANGELOG.md:1]:
```suggestion
# 2.14.0 / 2024-09-25
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+
+* [FEATURE] Add stop and start APIs for Session Replay. See [#2169](https://github.com/DataDog/dd-sdk-android/pull/2169)
```
Comment (0xnm) [inline CHANGELOG.md:3]: for the readability it is better to group categories: first `FEATURE`, then `IMPROVEMENT`, then `MAINTENANCE`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+* [MAINTENANCE] Make image privacy fine grained masking API public in Session Replay. See [#2204](https://github.com/DataDog/dd-sdk-android/pull/2204)
+* [FEATURE] Add touch privacy fine grained masking API to Session Replay. See [#2196](https://github.com/DataDog/dd-sdk-android/pull/2196)
+* [MAINTENANCE] Update benchmark metrics memory reader probe interval. See [#2228](https://github.com/DataDog/dd-sdk-android/pull/2228)
+* [IMPROVEMENT] Enable Kotlin test fixtures support. See [#2234](https://github.com/DataDog/dd-sdk-android/pull/2234)
+* [FEATURE] Add text and input privacy fine grained masking API to Session Replay. See [#2235](https://github.com/DataDog/dd-sdk-android/pull/2235)
+* [MAINTENANCE] Fix the flakiness in the KioskTrackingTest. See [#2226](https://github.com/DataDog/dd-sdk-android/pull/2226)
```
Comment (0xnm) [inline CHANGELOG.md:9]:
```suggestion
* [MAINTENANCE] Fix the flakiness in the `KioskTrackingTest`. See [#2226](https://github.com/DataDog/dd-sdk-android/pull/2226)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,32 @@
...
+* [IMPROVEMENT] Add isContainer attribute to session replay span. See [#2244](https://github.com/DataDog/dd-sdk-android/pull/2244)
+* [MAINTENANCE] Fix placeholder dimensions. See [#2248](https://github.com/DataDog/dd-sdk-android/pull/2248)
+* [MAINTENANCE] Send fine grained masking instead of legacy privacy in config telemetry. See [#2253](https://github.com/DataDog/dd-sdk-android/pull/2253)
+* [MAINTENANCE] ensure UploadWorker uses the sdk instance name. See [#2257](https://github.com/DataDog/dd-sdk-android/pull/2257)
```
Comment (0xnm) [inline CHANGELOG.md:13]:
```suggestion
* [MAINTENANCE] Ensure `UploadWorker` uses the SDK instance name. See [#2257](https://github.com/DataDog/dd-sdk-android/pull/2257)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,32 @@
+# 2.14.0 / 2024-09-24
+
+* [FEATURE] Add stop and start APIs for Session Replay. See [#2169](https://github.com/DataDog/dd-sdk-android/pull/2169)
+* [MAINTENANCE] Make image privacy fine grained masking API public in Session Replay. See [#2204](https://github.com/DataDog/dd-sdk-android/pull/2204)
+* [FEATURE] Add touch privacy fine grained masking API to Session Replay. See [#2196](https://github.com/DataDog/dd-sdk-android/pull/2196)
+* [MAINTENANCE] Update benchmark metrics memory reader probe interval. See [#2228](https://github.com/DataDog/dd-sdk-android/pull/2228)
+* [IMPROVEMENT] Enable Kotlin test fixtures support. See [#2234](https://github.com/DataDog/dd-sdk-android/pull/2234)
+* [FEATURE] Add text and input privacy fine grained masking API to Session Replay. See [#2235](https://github.com/DataDog/dd-sdk-android/pull/2235)
+* [MAINTENANCE] Fix the flakiness in the KioskTrackingTest. See [#2226](https://github.com/DataDog/dd-sdk-android/pull/2226)
+* [IMPROVEMENT] Add isContainer attribute to session replay span. See [#2244](https://github.com/DataDog/dd-sdk-android/pull/2244)
```
Comment (0xnm) [inline CHANGELOG.md:10]:
```suggestion
* [IMPROVEMENT] Add `isContainer` attribute to session replay span. See [#2244](https://github.com/DataDog/dd-sdk-android/pull/2244)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,32 @@
...
+* [MAINTENANCE] Fix placeholder dimensions. See [#2248](https://github.com/DataDog/dd-sdk-android/pull/2248)
+* [MAINTENANCE] Send fine grained masking instead of legacy privacy in config telemetry. See [#2253](https://github.com/DataDog/dd-sdk-android/pull/2253)
+* [MAINTENANCE] ensure UploadWorker uses the sdk instance name. See [#2257](https://github.com/DataDog/dd-sdk-android/pull/2257)
+* [MAINTENANCE] Explicitly set `antlr-runtime` transitive dependency version. See [#2261](https://github.com/DataDog/dd-sdk-android/pull/2261)
+* [FEATURE] Introduce the RumMonitor#addViewLoadingTime API. See [#2243](https://github.com/DataDog/dd-sdk-android/pull/2243)
```
Comment (0xnm) [inline CHANGELOG.md:15]:
```suggestion
* [FEATURE] Introduce the `RumMonitor#addViewLoadingTime` API. See [#2243](https://github.com/DataDog/dd-sdk-android/pull/2243)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,32 @@
...
+* [FEATURE] Introduce the API usage telemetry event and API. See [#2258](https://github.com/DataDog/dd-sdk-android/pull/2258)
+* [IMPROVEMENT] Update custom detekt CI Job. See [#2118](https://github.com/DataDog/dd-sdk-android/pull/2118)
+* [IMPROVEMENT] Randomize privacy levels to support Fine Grained Masking in E2E. See [#2265](https://github.com/DataDog/dd-sdk-android/pull/2265)
+* [IMPROVEMENT] Update AGP to 8.6.1. See [#2269](https://github.com/DataDog/dd-sdk-android/pull/2269)
+* [IMPROVEMENT] Add telemetry and logs related with RumMonitor#addViewLoadingTime API. See [#2267](https://github.com/DataDog/dd-sdk-android/pull/2267)
```
Comment (0xnm) [inline CHANGELOG.md:20]:
```suggestion
* [IMPROVEMENT] Add telemetry and logs related with `RumMonitor#addViewLoadingTime` API. See [#2267](https://github.com/DataDog/dd-sdk-android/pull/2267)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,32 @@
...
+* [MAINTENANCE] Add the integration tests related with RumMonitor#addViewLoadingTime API. See [#2268](https://github.com/DataDog/dd-sdk-android/pull/2268)
```
Comment (0xnm) [inline CHANGELOG.md:21]:
```suggestion
* [MAINTENANCE] Add the integration tests related with `RumMonitor#addViewLoadingTime` API. See [#2268](https://github.com/DataDog/dd-sdk-android/pull/2268)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,32 @@
...
+* [IMPROVEMENT] Handle SSE requests. See [#2270](https://github.com/DataDog/dd-sdk-android/pull/2270)
+* [IMPROVEMENT] Do not use magic numbers in InternalLogger API. See [#2271](https://github.com/DataDog/dd-sdk-android/pull/2271)
```
Comment (0xnm) [inline CHANGELOG.md:23]:
```suggestion
* [IMPROVEMENT] Do not use magic numbers in `InternalLogger` API. See [#2271](https://github.com/DataDog/dd-sdk-android/pull/2271)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,32 @@
...
+* [IMPROVEMENT] Optimize MD5 byte array to hex string conversion. See [#2273](https://github.com/DataDog/dd-sdk-android/pull/2273)
+* [MAINTENANCE] Fix DatadogInterceptor flaky test. See [#2274](https://github.com/DataDog/dd-sdk-android/pull/2274)
```
Comment (0xnm) [inline CHANGELOG.md:25]:
```suggestion
* [MAINTENANCE] Fix `DatadogInterceptor` flaky test. See [#2274](https://github.com/DataDog/dd-sdk-android/pull/2274)
```

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

**Pull request merged**